.github/workflows/build_and_test.yml

```diff
@@ -82,7 +82,6 @@ jobs:
     secrets: inherit
 
   test-mac-main-everest:
-    if: github.ref == 'refs/heads/main' # only perform mac tests on main branch
```

Comment:
I don't think we want to remove this line now

Reply:
This PR was only for the sake of testing MacOS runners on PR.